test/SmokeTests/OOProcSmokeTests/DotNetIsolated/FaultyOrchestrators.cs

```diff
@@ -0,0 +1,155 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Client;
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace FaultOrchestrators
+{
+    public static class FaultyOrchestrators
+    {
+        [Function(nameof(OOMOrchestrator))]
+        public static Task OOMOrchestrator(
+            [OrchestrationTrigger] TaskOrchestrationContext context)
+        {
+            // this orchestrator is not deterministic, on purpose.
+            // we use the non-determinism to force an OOM exception on only the first replay
+            
+            // check if a file named "replayEvidence" exists in source code directory, create it if it does not.
+            // From experience, this code runs in `<sourceCodePath>/bin/output/`, so we store the file two directories above.
+            // We do this because the /bin/output/ directory gets overridden during the build process, which happens automatically
+            // when `func host start` is re-invoked.
+            string evidenceFile = System.IO.Path.Combine(System.IO.Directory.GetCurrentDirectory(), "..", "..", "replayEvidence");
+            bool isTheFirstReplay = !System.IO.File.Exists(evidenceFile);
+            if (isTheFirstReplay)
+            {
+                System.IO.File.Create(evidenceFile).Close();
+
+                // force the process to run out of memory
+                List<byte[]> data = new List<byte[]>();
+
+                for (int i = 0; i < 10000000; i++)
+                {
+                    data.Add(new byte[1024 * 1024 * 1024]);
+                }
+            }
+            
+            // assuming the orchestrator survived the OOM, delete the evidence file and return
```

Comment:
I'm not sure I understand this comment, and it's making me question whether I understand what this test is supposed to be doing. I'll ask simply: why would we expect an orchestrator to survive an OOM? It sounds like you're saying there was an OOM but it was handled and we kept running in spite of the OOM. Are you actually saying it *recovered from a worker process crash* (caused by the OOM)? Maybe OOM implies crash, but it's not 100% clear that's what you meant.

Reply:
You're right the wording here was not great.

I do not expect an orchestration invocation to continue running beyond an OOM.

What I mean by survive was:
- orchestrator invocation A runs
- invocation A OOMs
- DF extension abandons the invocation
- DF extension retries the orchestrator invocation (now invocation B, every invocation is unique)
- invocation does not OOM now, and completes

So it "survived"  because it retried.

I'm hoping the new comments here make this clearer: https://github.com/Azure/azure-functions-durable-extension/pull/2922/commits/a70d0146fae6771f6cfd834a374407045043a9ec